frontera/contrib/scrapy/schedulers/frontier.py

```diff
@@ -124,11 +134,13 @@ def process_exception(self, request, exception, spider):
     def open(self, spider):
         log.msg('Starting frontier', log.INFO)
         if not self.frontier.manager.auto_start:
-            self.frontier.start()
+            scrapy_kwargs = {'spider': spider}
+            self.frontier.start(**scrapy_kwargs)
```

Comment:
«Regarding third one, frontier parameters. May be it's better to use Frontera»

As I said in the comment, they are dynamic and cannot be set via hardcoded settings. They are job dependent, so they are passed as spider argument. And in order to access the spider arguments, we need to access the spider.